lib/spack/llnl/util/symlink.py

```diff
@@ -30,9 +30,15 @@ def symlink(real_path, link_path):
         try:
             # Try to use junctions
             _win32_junction(real_path, link_path)
-        except OSError:
-            # If all else fails, fall back to copying files
-            shutil.copyfile(real_path, link_path)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                # we manually raised this error, we know shutil will fail the copy
+                # due to a same file error
+                # skip and report
```

Comment:
IMO it's more important that the error explains that we don't need to perform this operation, vs. explaining that we know it was raised manually (I would avoid that kind of explanation in general, including use of the term "manual")

Reply:
I'll refactor the explanation along those lines.